src/index.js

```diff
@@ -35,169 +119,204 @@ class QueuedTask extends HasModels {
       modelNames: {
         'queued_task': 'QueuedTask' }
     })
+    this._queues = {}
+    application.once('launch', async () => {
+      let rslts = await this.models.QueuedTask.destroy({expiresAt: {'<': new Date()}})
+      if (rslts.length > 0)
+        this.log.debug(`deleted ${rslts.length} expired queued tasks`)
+    })
   }
 
-  /** Creates a queued task.
-   * @param {string|Object} properties initial properties; may be either
-   *   a `name` string or an object containing `name`, `progress` and
-   *   `taskData` properties.
-   * @return {Object} created queued task entity
+  /** Creates a task queue.
+   * This is a simple wrapper around a nxus-worker-queue task handler
+   * that adapts it to use task state objects.
+   *
+   * @param {string} name - name of the task queue
+   * @param {Function} handler - handler for processing task requests;
+   *   passed a task state object; should return a promise that resolves
+   *   on completion to an object containing task state properties
+   *   (suitable for input to `updateTaskStatus()`)
+   * @param {Object} options - queue options (passed to the Bull
+   *   `Queue` constructor)
    */
-  async createTask(properties) {
-    if (typeof properties === 'string') properties = {name: properties}
-    properties = Object.assign({}, defaultTaskProperties, pick(properties, initialTaskProperties))
-    properties.completed = properties.progress >= 1.0
-    let rslt = await this.models.QueuedTask.create(properties)
-    return rslt
+  createTaskQueue(name, handler, options = {}) {
+    workerQueue.worker(name,
+      async (job) => {
```

Comment:
Reasonable to have this as an anon function, but probably makes it so this job handling wrapper can't be tested outside of (or without mocking) workerQueue.

Reply:
I don't understand your comment. `createTaskQueue()` is what you call to initialize a task queue (instead of doing setup at the `workerQueue` level). So it wouldn't make sense to be anon. Am I missing something?

Reply:
Sorry, was referring to the `async (job)` anon internal handler here -`createTaskQueue` is fine - where I'd consider having unit tests of the state initialization/update logic this module is layering around the handler. But not directly a concern.